src/main/java/io/appium/java_client/pagefactory/AppiumElementLocator.java

```diff
@@ -61,29 +59,21 @@
      *                          are found once should be cached
      * @param duration          is a POJO which contains timeout parameters for the element to be searched
      * @param originalDuration  is a POJO which contains timeout parameters from page object which contains the element
-     * @param originalWebDriver is an instance of WebDriver that is going to be
-     *                          used by a proxied element
      */
 
     public AppiumElementLocator(SearchContext searchContext, By by, boolean shouldCache,
-        TimeOutDuration duration, TimeOutDuration originalDuration, WebDriver originalWebDriver) {
+        TimeOutDuration duration, TimeOutDuration originalDuration) {
```

Comment:
would it be possible to also replace the custom TimeOutDuration class with the native Duration one, since we anyway change method signature?

Reply:
@mykola-mokhnach 
I think these parameters are needed still.

TimeOutDuration is just the container of duration values. 
https://github.com/appium/java-client/blob/master/src/main/java/io/appium/java_client/pagefactory/TimeOutDuration.java

It was designed for the purposes like 
https://github.com/TikhomirovSergey/java-client/blob/73077776d513e4521d7d5e8304469a4b85232e05/src/test/java/io/appium/java_client/pagefactory_tests/TimeOutTest.java#L122

I think it is more useful to completely change value than do 'plus' or 'minus'. 
However, I think I could redesign `TimeOutDuration` and make it work with `java.time.Duration`. I think it should be done step by step. The first is to mark some things `@Deprecated` and to add new good things instead, the second is removal. 

I just think that the change of TimeOutDuration may impact many users.

Reply:
yep, I agree about we can do a separate PR for it